OpenRA.Mods.Common/Activities/DeployForGrantedCondition.cs

```diff
@@ -64,4 +80,40 @@ public override Activity Tick(Actor self)
 			return NextActivity;
 		}
 	}
+
+	public class DeployInner : Activity
+	{
+		readonly GrantConditionOnDeploy deployment;
+		readonly bool towardDeploy;
+		bool initiated;
+
+		public DeployInner(Actor self, GrantConditionOnDeploy deployment, bool towardDeploy)
+		{
+			this.deployment = deployment;
+			this.towardDeploy = towardDeploy;
+
+			// Once deployment animation starts, the animation must finish.
+			IsInterruptible = false;
+		}
+
+		public override Activity Tick(Actor self)
+		{
+			// Wait for deployment
+			if (deployment.DeployState == DeployState.Deploying || deployment.DeployState == DeployState.Undeploying)
+				return this;
+
+			if (!initiated)
+			{
+				if (towardDeploy)
```

Comment:
Can't you just look at the `DeployState` and figure out from that what to do?

Reply:
Hmm, it seems that way, yes. I can't now remember what the reasoning behind that was.

Reply:
Is this comment still valid?

Reply:
Yep. Fixed it now.